tests/functional/functional-tests.js

```diff
@@ -25,20 +25,22 @@ import * as url from 'node:url'
 import async from 'async'
 import chai from 'chai'
 import _ from 'lodash'
+import { AssumeRoleProvider } from 'minio/dist/main/AssumeRoleProvider.js'
+import { CopyDestinationOptions, CopySourceOptions, DEFAULT_REGION } from 'minio/dist/main/helpers.js'
+import * as minio from 'minio/dist/main/minio.js'
```

Comment:
We should start importing the source code `minio/src/main/minio.ts` instead of importing the `dist` output.

Should adjust `mocha` to able to parse typescript files too, though.